File: app/src/main/java/han/ayeon/searchimgandvideo/util/SavedListLiveData.java
package han.ayeon.searchimgandvideo.util;

import androidx.lifecycle.LiveData;

import java.util.ArrayList;

import han.ayeon.searchimgandvideo.model.data.Media;

public class SavedListLiveData extends LiveData<ArrayList<Media>> implements ListLiveData {

    private ArrayList<Media> savedItemList = new ArrayList<>();

    public SavedListLiveData() {
        setValue(savedItemList);
    }

    @Override
    public void add(Media item) {
        savedItemList.add(item);
        this.setValue(savedItemList);
    }

    @Override
    public void clear() {
        savedItemList.clear();
        this.setValue(savedItemList);
    }

    @Override
    public void remove(Media item) {
        savedItemList.remove(item);
        this.setValue(savedItemList);
    }

}

File: app/src/main/java/han/ayeon/searchimgandvideo/util/SavedListLiveData.kt
package han.ayeon.searchimgandvideo.util

import androidx.lifecycle.LiveData
import han.ayeon.searchimgandvideo.model.data.Media
import java.util.ArrayList

/**
 * Created by HanAYeon on 2019-07-18.
 */

class SavedListLiveData : LiveData<ArrayList<Media>>(), ListLiveData {

    private val savedItemList = ArrayList<Media>()

    init {
        value = savedItemList
    }

    override fun add(item: Media) {
        savedItemList.add(item)
        this.value = savedItemList
    }

    override fun clear() {
        savedItemList.clear()
        this.value = savedItemList
    }

    override fun remove(item: Media) {
        savedItemList.remove(item)
        this.value = savedItemList
    }

}
